### PR TITLE
Best/worst offense tt

### PR DESCRIPTION
**Pull request by tylertomlinson**
Iteration 3 Best and Worse offense methods including helper methods.
All tests pass.

### CODE REVIEW

File: lib/game_teams_collection.rb [hunk 1]
```diff
@@ -15,6 +15,42 @@ def create_game_teams_array(file_path)
...
+    total_wins_per_team / total_games_per_team(team_id).to_f
+  end
+
+  def total_goals_by_team_id(team_id)
+    games_by_team_id(team_id).sum {|game_team| game_team.goals.to_i}
+  end
+
+  def average_goals_per_team_id(team_id)
+    (total_goals_by_team_id(team_id).to_f / games_by_team_id(team_id).count).to_f
+  end
```
Comment (mronauli) [inline lib/game_teams_collection.rb]: possible module for average?

File: lib/game_teams_collection.rb [hunk 2]
```diff
@@ -15,6 +15,42 @@ def create_game_teams_array(file_path)
     load_from_csv(file_path, GameTeams)
   end
 
+  def games_by_team_id(team_id)
+    @game_teams_array.select {|game_team| game_team.team_id.to_i == team_id}
+  end
+
+  def total_wins_per_team
+    @game_teams_array.select {|game_team| game_team.result == "WIN"}.count
+  end
```
Comment (mronauli) [inline lib/game_teams_collection.rb]: code looks clean

File: lib/game_teams_collection.rb [hunk 3]
```diff
@@ -33,4 +33,8 @@ def average_win_percentage(team_id)
   def total_goals_by_team_id(team_id)
     games_by_team_id(team_id).sum {|game_team| game_team.goals.to_i}
   end
+
+  def average_goals_per_team_id(team_id)
+    (total_goals_by_team_id(team_id).to_f / games_by_team_id(team_id).count).to_f
```
Comment (Yetidancer) [inline lib/game_teams_collection.rb]: I think your to_f here is redundant

Comment (tylertomlinson) [reply lib/game_teams_collection.rb]: Changed. Thanks. Learned something!

File: lib/game_teams_collection.rb [hunk 4]
```diff
@@ -15,6 +15,42 @@ def create_game_teams_array(file_path)
     load_from_csv(file_path, GameTeams)
   end
 
+  def games_by_team_id(team_id)
+    @game_teams_array.select {|game_team| game_team.team_id.to_i == team_id}
```
Comment (d-atkins) [inline lib/game_teams_collection.rb]: I would change game_teams' team_id attribute to an i so you don't have to .to_i it every time

**Review by d-atkins — APPROVED** [review]
it's good now

File: lib/game_teams_collection.rb (hunk 4 above)
Comment (tylertomlinson) [reply lib/game_teams_collection.rb]: Good catch

File: lib/game_teams_collection.rb [hunk 5]
```diff
@@ -15,6 +15,42 @@ def create_game_teams_array(file_path)
     load_from_csv(file_path, GameTeams)
   end
 
+  def games_by_team_id(team_id)
+    @game_teams_array.select {|game_team| game_team.team_id == team_id}
+  end
+
+  def total_wins_per_team
+    @game_teams_array.select {|game_team| game_team.result == "WIN"}.count
+  end
```
Comment (mronauli) [inline lib/game_teams_collection.rb:24]: looks clean

File: lib/game_teams_collection.rb (hunk 1 above)
Comment (tylertomlinson) [reply lib/game_teams_collection.rb]: Yes I think so.

File: lib/game_teams_collection.rb (hunk 5 above)
Comment (tylertomlinson) [reply lib/game_teams_collection.rb:24]: MR. CLEAN

**Pull request merged**